src/caf/core/data_structures.py

```diff
@@ -684,8 +870,564 @@ def aggregate(self, segs: list[str]):
+            diff = (
+                check.aggregate(target.data.segmentation, _bypass_validation=True).__sub__(
+                    target.data, _bypass_validation=True
+                )
+            ) ** 2
+            mse += diff.sum() / len(target.data)
+        return mse**0.5
+
+    def validate_ipf_targets(self, targets: list[IpfTarget]):
+        """
+        Check targets for ipf will work, raises errors if not.
+
+        Parameters
+        ----------
+        targets: list[IpfTarget]
+            List of IPF targets to validate.
+
+        Returns
+        -------
+        Input targets, with zone translations added in if relevant.
+        """
+        target_sum = 0
+        for position, target in enumerate(targets):
+            # Check targets sum to the same, or they can't converge. Potentially could allow
+            # IPF for non-agreeing targets to get as close as possible.
+            zoning_diff = False
+            if target_sum == 0:
+                target_sum = target.data.sum()
+            else:
+                # TODO don't hard code this
+                if not math.isclose(target_sum, target.data.sum(), abs_tol=target_sum / 1e5):
+                    raise ValueError(
+                        "Input target DVectors do not have consistent "
+                        "sums, so ipf will fail."
+                    )
+            # Check segmentations are compatible.
+            if not target.data.segmentation.is_subset(self.segmentation):
+                if target.segment_translations is None:
+                    raise ValueError(
+                        "The target segmentation is not a subset of the seed "
+                        "segmentation, but no correspondences are defined."
+                    )
+                non_matching = target.data.segmentation - self.segmentation
+                for seg in non_matching:
+                    seg = seg.name
+                    if seg in target.segment_translations.keys():
+                        if target.segment_translations[seg] in self.segmentation.names:
+                            lower_seg = self.segmentation.get_segment(
+                                target.segment_translations[seg]
+                            )
+                            try:
+                                # Don't need this for now, checking it exists.
+                                lower_seg.translate_segment(seg)
+                            except FileNotFoundError:
+                                raise FileNotFoundError(
+                                    f"No segment translation found for {lower_seg} to {seg}.from."
+                                )
+
+            # Check zoning systems are compatible.
+            if self.zoning_system != target.data.zoning_system:
+                zoning_diff = True
+                if target.zone_translation is not None:
+                    pass
+                else:
```

Comment:
If nothing is happening in the if then this could be simplified
```suggestion
                if target.zone_translation is None:
```